tests/backends/data/test_async_mongo.py

```diff
@@ -0,0 +1,1087 @@
+    ]
+    assert result == expected[:2]
+    result = [
+        statement async for statement in backend.read(raw_output=True, chunk_size=2)
+    ]
+    assert result == expected
+    result = [
+        statement async for statement in backend.read(raw_output=True, chunk_size=1000)
+    ]
+    assert result == expected
+
+
+@pytest.mark.anyio
+async def test_backends_data_async_mongo_data_backend_read_method_without_raw_output(
+    mongo, async_mongo_backend
+):
+    """Test the `AsyncMongoDataBackend.read` method with `raw_output` set to
+    `False`.
+    """
+    # pylint: disable=unused-argument
+    backend = async_mongo_backend()
+    documents = [
+        {"_id": ObjectId("64945e53a4ee2699573e0d6f"), "id": "foo"},
+        {"_id": ObjectId("64945e530468d817b1f756da"), "id": "bar"},
+        {"_id": ObjectId("64945e530468d817b1f756db"), "id": "baz"},
+    ]
+    expected = [
+        {"_id": "64945e53a4ee2699573e0d6f", "id": "foo"},
+        {"_id": "64945e530468d817b1f756da", "id": "bar"},
+        {"_id": "64945e530468d817b1f756db", "id": "baz"},
+    ]
+    await backend.collection.insert_many(documents)
+    await backend.database.foobar.insert_many(documents[:2])
+
+    assert [statement async for statement in backend.read()] == expected
+    assert [statement async for statement in backend.read(target="foobar")] == expected[
+        :2
+    ]
+    assert [statement async for statement in backend.read(chunk_size=2)] == expected
+    assert [statement async for statement in backend.read(chunk_size=1000)] == expected
+
+
+@pytest.mark.parametrize(
+    "invalid_target,error",
+    [
+        (".foo", "must not start or end with '.': '.foo'"),
+        ("foo.", "must not start or end with '.': 'foo.'"),
+        ("foo$bar", "must not contain '$': 'foo$bar'"),
+        ("foo..bar", "cannot be empty"),
+    ],
+)
+@pytest.mark.anyio
+async def test_backends_data_async_mongo_data_backend_read_method_with_invalid_target(
+    mongo,  # pylint: disable = unused-argument
+    invalid_target,
+    error,
+    async_mongo_backend,
+    caplog,  # pylint: disable=unused-argument
+):
+    """Test the `AsyncMongoDataBackend.read` method given an invalid `target` argument,
+    should raise a `BackendParameterException`.
+    """
+    backend = async_mongo_backend()
+    msg = (
+        f"The target=`{invalid_target}` is not a valid collection name: "
+        f"collection names {error}"
+    )
+    with pytest.raises(BackendParameterException, match=msg.replace("$", r"\$")):
+        with caplog.at_level(logging.ERROR):
+            async for statement in backend.read(target=invalid_target):
+                next(statement)
+
+    assert (
+        "ralph.backends.data.async_mongo",
+        logging.ERROR,
+        msg,
+    ) in caplog.record_tuples
+
+
+@pytest.mark.anyio
+async def test_backends_data_async_mongo_data_backend_read_method_with_failure(
+    mongo, async_mongo_backend, monkeypatch, caplog  # pylint: disable=unused-argument
```

Comment:
```suggestion
    async_mongo_backend, monkeypatch, caplog
```
As the `find` method is mocked